src/components/HeartRateZoneGraph.vue

```diff
@@ -265,13 +266,15 @@ export default {
 
       // Process each data point and calculate heart rate zone and total time
       sessions.forEach((session) => {
-        if (!session.records) return
+        if (session.records == null) return
         for (let i = 0; i < session.records.length - 1; i++) {
           const entry = session.records[i]
           const nextEntry = session.records[i + 1]
 
-          const hrZoneIndex = this.getHeartRateZoneIndex(entry.heartRate || 0)
-          const nextHrZoneIndex = this.getHeartRateZoneIndex(nextEntry.heartRate || 0)
+          if (entry.heartRate == null || nextEntry.heartRate == null) continue
```

Comment:
If we have these records pattern, do we still get zone? It seems like will be skipped entirely, no? What if for nextEntry instead of [i+1], we try find up to next record with valid heartRate? Then we move our `i` cursor to that record index.

```
records[0].heartRate = 120
records[1].heartRate = null
records[2].heartRate = 130
records[3].heartRate = null
records[4].heartRate = 135
records[5].heartRate = null
```

Reply:
It will skipped entirely.
is it possible to have that case commonly?
if such case exists as rare, we can adjust how will we recognize this data case.

> What if for nextEntry instead of [i+1], we try find up to next record with valid heartRate?

If this process is valid, we can adjust to it.

Assume there a case
```
records[0].heartRate = 120
records[1].heartRate = null
records[2].heartRate = null
records[3].heartRate = null
records[4].heartRate = null
records[5].heartRate = 135
```
what should we do then? compare 120 and 135?

Reply:
> is it possible to have that case commonly?
We can't be sure since the data is from sensor, sometimes the sensor is not working properly due to low battery or possition, network fail between sensor to cyclocomp, etc.

> what should we do then? compare 120 and 135?
Yes, compare let's compare 120 to 135, and calculate the elapse between them as usual.

Reply:
Updated, latest commit now will do compare 120 and 135.
then next record will do 135 and next valid HR (if no valid HR, it will compare with themself for diff time 1s)

Reply:
Since i don't have invalid data, i'll do test manually via code. kindly help me check with your real data